tests/protocol/rwa-pool/Pool.Rwa.t.sol

```diff
@@ -0,0 +1,431 @@
+
+contract PoolRwaTests is TestnetProcedures {
+  using stdStorage for StdStorage;
+  using ReserveConfiguration for DataTypes.ReserveConfigurationMap;
+
+  function setUp() public virtual {
+    initTestEnvironment();
+  }
+
+  function test_reverts_mintToTreasury() public {
+    _seedLiquidity({token: tokenList.buidl, amount: 50_000e6, isRwa: true});
+    _setReserveBorrowingAndReserveFactor(tokenList.buidl);
+
+    // upgrade aBuidl to the standard aToken implementation, to be able to borrow
+    _upgradeToStandardAToken();
+
+    (, , address varDebtBuidl) = contracts.protocolDataProvider.getReserveTokensAddresses(
+      tokenList.buidl
+    );
+
+    vm.startPrank(bob);
+    contracts.poolProxy.supply(tokenList.wbtc, 0.4e8, bob, 0);
+    contracts.poolProxy.borrow(tokenList.buidl, 2000e6, 2, 0, bob);
+    skip(30 days);
+    contracts.poolProxy.repay(tokenList.buidl, IERC20Detailed(varDebtBuidl).balanceOf(bob), 2, bob);
+    vm.stopPrank();
+
+    // distribute fees to treasury
+    address[] memory assets = new address[](1);
+    assets[0] = tokenList.buidl;
+
+    // upgrade aBuidl to the rwa aToken implementation, to test that mintToTreasury reverts
+    _upgradeToRwaAToken();
+
+    // expect call by matching the selector only
+    vm.expectCall(rwaATokenList.aBuidl, abi.encodeWithSelector(IRwaAToken.mintToTreasury.selector));
+
+    vm.expectRevert(bytes(Errors.OPERATION_NOT_SUPPORTED));
+    contracts.poolProxy.mintToTreasury(assets);
+  }
+
+  function test_setUserUseReserveAsCollateral_false() public {
+    vm.startPrank(alice);
+    contracts.poolProxy.supply(tokenList.buidl, 1e6, alice, 0);
+
+    vm.expectEmit(address(contracts.poolProxy));
+    emit IPool.ReserveUsedAsCollateralDisabled(tokenList.buidl, alice);
+
+    contracts.poolProxy.setUserUseReserveAsCollateral(tokenList.buidl, false);
+    vm.stopPrank();
+  }
+
+  function test_setUserUseReserveAsCollateral_true() public {
+    test_setUserUseReserveAsCollateral_false();
+
+    vm.expectEmit(address(contracts.poolProxy));
+    emit IPool.ReserveUsedAsCollateralEnabled(tokenList.buidl, alice);
+
+    vm.prank(alice);
+    contracts.poolProxy.setUserUseReserveAsCollateral(tokenList.buidl, true);
+  }
+
+  function test_noop_setUserUseReserveAsCollateral_true_when_already_is_activated() public {
+    test_setUserUseReserveAsCollateral_true();
+
+    vm.prank(alice);
+    contracts.poolProxy.setUserUseReserveAsCollateral(tokenList.buidl, true);
+  }
+
+  function test_reverts_setUserUseReserveAsCollateral_true_ltv_zero() public {
+    test_setUserUseReserveAsCollateral_false();
+
+    vm.prank(poolAdmin);
+    contracts.poolConfiguratorProxy.configureReserveAsCollateral(tokenList.buidl, 0, 70_00, 105_00);
+
+    vm.expectRevert(bytes(Errors.USER_IN_ISOLATION_MODE_OR_LTV_ZERO));
+
+    vm.prank(alice);
+    contracts.poolProxy.setUserUseReserveAsCollateral(tokenList.buidl, true);
+  }
+
```

Comment:
did we not include test_reverts_setUserUseReserveAsCollateral_true_user_isolation_mode intentionally?